### ISSUE TITLE
Support RTP extensions properly

### ISSUE THREAD

**Issue by mickel8** [membraneframework/membrane_rtp_plugin#23]
It seems that we don't handle RTP extensions properly. If they are set (in SDP offer/answer) our depayloaders receive improper payloads. This happens both for OPUS and H264 depayloaders. Removing RTP extensions from SDP answer seems to resolve problem. 

Example extension set that causes errors in OPUS (audio):

```
    a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
    a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
    a=extmap:3 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
    a=extmap:4 urn:ietf:params:rtp-hdrext:sdes:mid
    a=extmap:5 urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id
    a=extmap:6 urn:ietf:params:rtp-hdrext:sdes:repaired-rtp-stream-id
```

Example extension set that causes errors in H264 (video):

```
    a=extmap:14 urn:ietf:params:rtp-hdrext:toffset
    a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
    a=extmap:13 urn:3gpp:video-orientation
    a=extmap:3 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
    a=extmap:12 http://www.webrtc.org/experiments/rtp-hdrext/playout-delay
    a=extmap:11 http://www.webrtc.org/experiments/rtp-hdrext/video-content-type
    a=extmap:7 http://www.webrtc.org/experiments/rtp-hdrext/video-timing
    a=extmap:8 http://www.webrtc.org/experiments/rtp-hdrext/color-space
    a=extmap:4 urn:ietf:params:rtp-hdrext:sdes:mid
    a=extmap:5 urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id
    a=extmap:6 urn:ietf:params:rtp-hdrext:sdes:repaired-rtp-stream-id
```

The task is to investigate where is the problem.

**Comment by mat-hek** [comment]
please share a sample packet that fails to be parsed

**Comment by mickel8** [comment]
Sure, I will share example packets that should be parsed properly but on Monday morning if it's not a problem

**Comment by mickel8** [comment]
So when appending the above extensions for video we can't depayload following packets:

```elixir
12:51:53.310 [warn]  [:depayloader] An error occurred while parsing RTP packet.
Reason: malformed_data
Packet: %Membrane.Buffer{metadata: %{rtp: %{payload_type: 108, sequence_number: 30873, ssrc: 1130350827, timestamp: 2415657469}, timestamp: 0}, payload: <<160, 49, 0, 1, 64, 49, 0, 0, 0, 103, 66, 192, 31, 140, 141, 64, 80, 30, 144, 15, 8, 132, 106>>}
```

or

```elixir
12:51:53.321 [warn]  [:depayloader] An error occurred while parsing RTP packet.
Reason: malformed_data
Packet: %Membrane.Buffer{metadata: %{rtp: %{payload_type: 108, sequence_number: 30875, ssrc: 1130350827, timestamp: 2415657469}, timestamp: 0}, payload: <<160, 49, 0, 3, 64, 49, 0, 0, 0, 101, 184, 0, 4, 0, 0, 12, 228, 197, 0, 1, 7, 169, 57, 57, 56, 99, 13, 41, 159, 26, 185, 85, 115, 234, 174, 85, 92, 77, 248, 237, 202, 219, 130, 60, 112, 0, 16, 59, 210, 114, 114, 127, 240, 255, 130, 204, 7, 181, 166, 109, 16, 0, 139, 248, 28, 34, 145, 114, 28, 34, 145, 112, 71, 227, 128, 0, 130, 245, 39, 39, 39, 8, 99, 73, 159, 141, 92, 170, 185, 245, 87, 42, 174, 8, 203, 24, 224, 0, 32, 159, 73, 201, 201, 255, 195, 254, 11, 48, 30, 211, 76, 232, 128, 0, 84, 27, 240, 56, 0, 130, 11, 185, 14, 0, 32, 130, 238, 8, 207, 12, 112, 0, 16, 114, 142, 100, 228, 228, 225, 12, 52, 41, 159, 26, 185, 85, 115, 234, 174, 85, 92, 17, 150, 199, 0, 1, 4, 250, 78, 78, 79, 254, 31, 240, 89, 128, 246, 154, 103, 68, 0, 34, 254, 7, 8, 164, 92, 135, 8, 164, 92, 17, 159, 142, 0, 2, 12, 26, 78, 78, 78, 24, 195, 74, 103, 198, 174, 85, 92, 250, 171, 149, 87, 19, 126, 59, 114, 182, 224, 140, 182, 56, 0, 8, 28, 233, 57, 57, 57, 58, 235, 174, 186, 122, 122, 235, 174, 158, 158, 186, 235, 173, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 116, 245, 215, 93, 61, 61, 117, 215, 79, 79, 93, 117, 215, 93, 117, 215, 79, 79, 93, 117, 211, 211, 215, 93, 117, 211, 215, 93, 116, 244, 245, 215, 93, 120>>}
```

**Comment by mickel8** [comment]
In case of Opus, depayloader actually doesn't complain (probably because it doesn't affect packets) but decoder can't decode packets e.g.

```elixir
%Membrane.Buffer{
  metadata: %{
    rtp: %{
      payload_type: 111,
      sequence_number: 28292,
      ssrc: 374705657,
      timestamp: 3812111525
    },
    timestamp: 0
  },
  payload: <<234, 185, 64, 48, 16, 196, 120, 1, 119, 214, 186, 72, 95, 201, 37,
    45, 26, 99, 186, 253, 105, 18, 193, 207, 155, 34, 164, 125, 45, 206, 10,
    119, 43, 150, 23, 47, 5, 171, 32, 95, 242, 80, 253, 53, 71, 48, 170, 91,
    112, 18, 69, 121, 191, 247, 182, 68, 145, 134, 142, 244, 248, 34, 141, 73,
    255>>
}
```
with the following error 
```elixir
Reason: {"corrupted stream", [{Membrane.Opus.Decoder.Native.Nif, :unifex_decode_packet, [#Reference<0.1442717731.1936326658.92199>, <<234, 185, 64, 48, 16, 196, 120, 1, 119, 214, 186, 72, 95, 201, 37, 45, 26, 99, 186, 253, 105, 18, 193, 207, 155, 34, 164, 125, 45, 206, 10, 119, 43, 150, 23, 47, 5, 171, 32, 95, 242, 80, ...>>], []},
```

**Comment by mat-hek** [comment]
This is fixed by #27, though [RFC 3550 section 5.3](https://tools.ietf.org/html/rfc3550#section-5.3) states that there may be RTP header extensions that are supposed to be handled by whatever what wants to get the actual payload. Such extensions are not put into the extension header, but just into the payload. Thus, some extensions may have to be turned off if we don't support them, or they will lead to 'malformed data' or 'corrupted stream' anyway.
